plugins/org.eclipse.mat.parser/src/org/eclipse/mat/parser/internal/snapshot/ObjectMarkerFactory.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2025 SAP AG and IBM Corporation.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    IBM
+ *******************************************************************************/
+package org.eclipse.mat.parser.internal.snapshot;
+
+import org.eclipse.mat.parser.index.IIndexReader;
+import org.eclipse.mat.util.IProgressListener;
+
+public class ObjectMarkerFactory
+{
+    private static final boolean useOldMarker = Boolean

```

Comment:
I think this is a great solution. We could even swap it to put the old marker as the default, and the new marker can be put aside until we can work out what the issue is.

Reply:
I was wondering about setting this to default until we can figure out what the issue is. You're okay with that?

Reply:
Yes, of course. It would make it difficult to collect data (since less users will run it), but, that's OK.